Source/WebCore/platform/graphics/egl/PlatformDisplaySurfaceless.cpp

```diff
@@ -33,11 +33,19 @@ namespace WebCore {
 
 std::unique_ptr<PlatformDisplaySurfaceless> PlatformDisplaySurfaceless::create()
 {
+    RefPtr<GLDisplay> glDisplay;
     const char* extensions = eglQueryString(nullptr, EGL_EXTENSIONS);
-    if (!GLContext::isExtensionSupported(extensions, "EGL_MESA_platform_surfaceless"))
+
+#if OS(ANDROID) && defined(EGL_PLATFORM_ANDROID_KHR)
+    // While not terribly common, custom Android builds other than AOSP or
+    // official Google ones may support to more than one window system.
+    if (GLContext::isExtensionSupported(extensions, "EGL_KHR_platform_android")) [[unlikely]]
```

Comment:
is EGL_KHR_platform_android surfaceless?

Reply:
After some discussion with @carlosgcampos, it seems better to add a `PlatformDisplayAndroid` class, so I'll rework things that way.